pkg/inventory/fake-inventory-client.go

```diff
@@ -57,8 +57,8 @@ func (fic *FakeInventoryClient) Merge(_ InventoryInfo, objs object.ObjMetadataSe
 
 // Replace the stored cluster inventory objs with the passed obj, or an
 // error if one is set up.
-
-func (fic *FakeInventoryClient) Replace(_ InventoryInfo, objs object.ObjMetadataSet, _ common.DryRunStrategy) error {
+func (fic *FakeInventoryClient) Replace(_ InventoryInfo, objs object.ObjMetadataSet, _ []ObjectStatus,
```

Comment:
The fact that this passes tests worries me. It seems to indicate we're missing tests.
We don't ever read the status back out again, so I suspect the gap is just in verifying that the status was stored.

Reply:
Added unit test for the `Replace` call.